### ISSUE TITLE
Bug in model.fit

### ISSUE THREAD

**Issue by JawadAr** [tensorflow/tensorflow#50348]
When calling model.fit in tensorflow version 2.5.0 the validation_data parameter doesn't work if the model takes 2 or more inputs. Same code in tensorflow 2.3.0 works perfectly fine. 

`history = model.fit(
	[X_train_pairs[:, 0], X_train_pairs[:, 1]], np.array(y_train),
  validation_data = ([X_val_pairs[:, 0], X_val_pairs[:, 1]], np.array(y_val)),
	batch_size=256,
	epochs=25)`

This is my code.

**Comment by tilakrayal** [comment]
@JawadAr ,

In order to expedite the trouble-shooting process, could you please provide the complete code and dataset to reproduce the issue reported here.
Thanks!

**Comment by JawadAr** [comment]
`def build_siamese_model(inputShape, embeddingDim=48):
	# specify the inputs for the feature extractor network
	inputs = Input(inputShape)

	# define the first set of CONV => RELU => POOL => DROPOUT layers
	x = Conv2D(64, (2, 2), padding="same", activation="relu")(inputs)
	x = MaxPooling2D(pool_size=(2, 2))(x)
	x = Dropout(0.3)(x)

	# second set of CONV => RELU => POOL => DROPOUT layers
	x = Conv2D(64, (2, 2), padding="same", activation="relu")(x)
	x = MaxPooling2D(pool_size=2)(x)
	x = Dropout(0.3)(x)

        # prepare the final outputs
	pooledOutput = GlobalAveragePooling2D()(x)
	outputs = Dense(embeddingDim)(pooledOutput)

	# build the model
	model = Model(inputs, outputs)

	# return the model to the calling function
	return model

def euclidean_distance(vectors):
	# unpack the vectors into separate lists
	(featsA, featsB) = vectors

	# compute the sum of squared distances between the vectors
	sumSquared = K.sum(K.square(featsA - featsB), axis=1,
		keepdims=True)

	# return the euclidean distance between the vectors
	return K.sqrt(K.maximum(sumSquared, K.epsilon()))


print("[INFO] building siamese network...")
imgA = Input(shape=config.IMG_SHAPE)
imgB = Input(shape=config.IMG_SHAPE)
featureExtractor = build_siamese_model(config.IMG_SHAPE)
featsA = featureExtractor(imgA)
featsB = featureExtractor(imgB)

distance = Lambda(utils.euclidean_distance)([featsA, featsB])
outputs = Dense(1, activation="sigmoid")(distance)
model = Model(inputs=[imgA, imgB], outputs=outputs)`

This is the code. The data is mnist and can be prepared using make_pairs function below:

`def make_pairs(images, labels):
	# initialize two empty lists to hold the (image, image) pairs and
	# labels to indicate if a pair is positive or negative
	pairImages = []
	pairLabels = []
	# calculate the total number of classes present in the dataset
	# and then build a list of indexes for each class label that
	# provides the indexes for all examples with a given label
	numClasses = len(np.unique(labels))
	idx = [np.where(labels == i)[0] for i in range(0, numClasses)]
	# loop over all images
	for idxA in range(len(images)):
		# grab the current image and label belonging to the current
		# iteration
		currentImage = images[idxA]
		label = labels[idxA]
		# randomly pick an image that belongs to the *same* class
		# label
		idxB = np.random.choice(idx[label])
		posImage = images[idxB]
		# prepare a positive pair and update the images and labels
		# lists, respectively
		pairImages.append([currentImage, posImage])
		pairLabels.append([1])
		# grab the indices for each of the class labels *not* equal to
		# the current label and randomly pick an image corresponding
		# to a label *not* equal to the current label
		negIdx = np.where(labels != label)[0]
		negImage = images[np.random.choice(negIdx)]
		# prepare a negative pair of images and update our lists
		pairImages.append([currentImage, negImage])
		pairLabels.append([0])
	# return a 2-tuple of our image pairs and labels
	return (np.array(pairImages), np.array(pairLabels))

# load MNIST dataset and scale the pixel values to the range of [0, 1]
print("[INFO] loading MNIST dataset...")
(trainX, trainY), (testX, testY) = mnist.load_data()
trainX = trainX / 255.0
testX = testX / 255.0
# add a channel dimension to the images
trainX = np.expand_dims(trainX, axis=-1)
testX = np.expand_dims(testX, axis=-1)
# prepare the positive and negative pairs
print("[INFO] preparing positive and negative pairs...")
(pairTrain, labelTrain) = utils.make_pairs(trainX, trainY)
(pairTest, labelTest) = utils.make_pairs(testX, testY)`

**Comment by JawadAr** [comment]
Please don't close it, work on it.

**Comment by tilakrayal** [comment]
@JawadAr ,

Looks like the code is incompleted.Please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/53eabc52f366dddd315ef7740fcd5cf1/untitled50348.ipynb).Can you please provide the complete code and dataset to trouble-shoot the issue.Thanks!

**Comment by JawadAr** [comment]
oh, I have to send the whole working code? The other part of the code is confidential by the client. I have sent all of the parts causing issue, if these are not enough I think the issue can be closed.

**Comment by tilakrayal** [comment]
@JawadAr ,

Please refer this link for more information on model.fit.It helps.[Link](https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit).Also please feel free to move this issue to closed status if issue has resolved.Thanks
